### DISCUSSION TITLE
Plugins from GHE and github.com private repo, possible?

### DISCUSSION THREAD

**Discussion by jaymecd** [terraform-linters/tflint#2004] [General]
Hi, I've faced a situation.
Context:
- tflint is trigger by pre-commit that run on CI pipeline
- there are 2 plugins: one hosted in corporate GitHub Enterprise, another - in github.com private repo
Since both APIs are protected with own token, using one `GITHUB_TOKEN` does not help much here.
Is there a way to overcome this limitation?
tnx

**Comment by wata727** [comment]
Thank you for reporting this. Unfortunately there is currently no way to support this situation with `tflint --init`.
A workaround is to install the plugins with your own script without using `tflint --init`.
See also https://github.com/terraform-linters/tflint/blob/v0.50.3/docs/user-guide/plugins.md#manual-installation

I've opened an issue tracking this problem, so please let us know if you are interested in a solution https://github.com/terraform-linters/tflint/issues/2005

**Reply by wata727** [reply]
In v0.51, you can set `GITHUB_TOKEN` per host, so this issue can be addressed by setting `GITHUB_TOKEN` for GitHub Enterprise hosts.
https://github.com/terraform-linters/tflint/blob/v0.51.0/docs/user-guide/plugins.md#avoiding-rate-limiting
